pkg/plugins/policies/meshratelimit/plugin/v1alpha1/plugin.go

```diff
@@ -25,6 +33,133 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
+	}
+	policies, ok := proxy.Policies.Dynamic[api.MeshRateLimitType]
+	if !ok {
+		return nil
+	}
+	listeners := policies_xds.GatherListeners(rs)
+	routes := policies_xds.GatherRoutes(rs)
+
+	if err := applyToInbounds(policies.FromRules, listeners.Inbound, proxy); err != nil {
+		return err
+	}
+	if err := applyToGateways(ctx, policies.FromRules, listeners.Gateway, routes.Gateway, proxy); err != nil {
+		return err
+	}
+	return nil
+}
+
+func applyToGateways(
+	ctx xds_context.Context,
+	fromRules core_xds.FromRules,
+	gatewayListeners map[core_xds.InboundListener]*envoy_listener.Listener,
+	gatewayRoutes map[string]*envoy_route.RouteConfiguration,
+	proxy *core_xds.Proxy,
+) error {
+	if !proxy.Dataplane.Spec.IsBuiltinGateway() {
+		return nil
+	}
+	gatewayListerInfos, err := gateway_plugin.GatewayListenerInfoFromProxy(context.TODO(), ctx.Mesh, proxy, ctx.ControlPlane.Zone)
+	if err != nil {
+		return err
+	}
+	for _, listenerInfo := range gatewayListerInfos {
+		address := proxy.Dataplane.Spec.GetNetworking().Address
+		port := listenerInfo.Listener.Port
+		listenerKey := core_xds.InboundListener{
+			Address: address,
+			Port:    port,
+		}
+		gatewayListener, ok := gatewayListeners[listenerKey]
+		if !ok {
+			continue
+		}
+		rules, ok := fromRules.Rules[listenerKey]
+		if !ok {
+			continue
+		}
+
+		route, ok := gatewayRoutes[listenerInfo.Listener.ResourceName]
+		if !ok {
+			continue
+		}
+
+		if err := configure(rules, proxy.Dataplane, gatewayListener, route); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func applyToInbounds(
+	fromRules core_xds.FromRules,
+	inboundListeners map[core_xds.InboundListener]*envoy_listener.Listener,
+	proxy *core_xds.Proxy,
+) error {
+	for _, inbound := range proxy.Dataplane.Spec.GetNetworking().GetInbound() {
+		iface := proxy.Dataplane.Spec.Networking.ToInboundInterface(inbound)
+		if _, exists := proxy.Policies.RateLimitsInbound[iface]; exists {
+			continue
+		}
+
+		listenerKey := core_xds.InboundListener{
+			Address: iface.DataplaneIP,
+			Port:    iface.DataplanePort,
+		}
+		listener, ok := inboundListeners[listenerKey]
+		if !ok {
+			continue
+		}
+		rules, ok := fromRules.Rules[listenerKey]
+		if !ok {
+			continue
+		}
+
+		if err := configure(rules, proxy.Dataplane, listener, nil); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func configure(
+	fromRules core_xds.Rules,
+	dataplane *core_mesh.DataplaneResource,
+	listener *envoy_listener.Listener,
+	route *envoy_route.RouteConfiguration,
+) error {
+	var conf api.Conf
+	// Currently, `from` section of MeshRateLimit only allows Mesh targetRef
+	if computed := fromRules.Compute(core_xds.MeshSubset()); computed != nil {
+		conf = computed.Conf.(api.Conf)
+	} else {
+		return nil
+	}
+
+	configurer := plugin_xds.Configurer{
+		Http: conf.Local.HTTP,
+		Tcp:  conf.Local.TCP,
+	}
+	log.V(1).Info(
+		"applying MeshRateLimit policies",
+		"dataplane", dataplane.Meta.GetName(),
+		"rules", conf,
+		"listener", listener,
+	)
```

Comment:
I'm hesitant about having this log even on debug. It can potentially produce many logs.
So far we were trying to avoid logging in `*generators` in "the old code" and I see no logs in other policies now.